packages/vx-demo/src/sandboxes/vx-chart-poc/src/types/context.ts

```diff
@@ -0,0 +1,109 @@
+import { UseTooltipParams } from '@vx/tooltip/lib/hooks/useTooltip';
+import { XYChartTheme } from './theme';
+import { ScaleType, Margin, ScaleOutput, LegendShape } from '.';
+
+// ChartContext ---------------------------------------------------------------
+export interface DataRegistry<Datum = unknown, XScaleInput = unknown, YScaleInput = unknown> {
+  [key: string]: {
+    /** unique data key */
+    key: string;
+    /** array of data */
+    data: Datum[];
+    /** function that returns the x value of a datum. */
+    xAccessor: (d: Datum) => XScaleInput;
+    /** function that returns the y value of a datum. */
+    yAccessor: (d: Datum) => YScaleInput;
+    /** whether the entry supports mouse events. */
+    mouseEvents?: boolean;
+    /** Optionally update the domain of the xScale. */
+    xDomain?: (domain: XScaleInput[]) => XScaleInput[];
+    /** Optionally update the domain of the yScale. */
+    yDomain?: (domain: YScaleInput[]) => YScaleInput[];
+    /** Optionally override logic for finding the nearest data point to a mouse event. */
+    findNearestDatum?: FindNearestDatum<Datum, XScaleInput, YScaleInput>;
+    /** Legend shape */
+    legendShape?: LegendShape;
+  };
+}
+
+export type RegisterData = (data: DataRegistry) => void;
+
+export type DatumWithKey<Datum = unknown> = { datum: Datum; key: string; index: number };
+
+export type FindNearestDatum<Datum, XScaleInput, YScaleInput> = (
+  args: NearestDatumArgs<Datum, XScaleInput, YScaleInput>,
+) => null | {
+  /** Closest datum. */
+  datum: Datum;
+  /** Index of the closest datum. */
+  index: number;
+  /** X coord distance in px from event to datum. Used to rank overall closest datum. */
+  distanceX: number;
+  /** Y coord distance in px from event to datum. Used to rank overall closest datum. */
+  distanceY: number;
+};
+
+export interface ChartContext<
+  Datum = unknown,
```

Comment:
Can this be `{ [key in DataKeys]: unknown }` ?
May have to change order of the generic type.